src/python_be.cc

```diff
@@ -2293,7 +2308,7 @@ TRITONBACKEND_Initialize(TRITONBACKEND_Backend* backend)
   return nullptr;
 }
 
-TRITONSERVER_Error*
+TRITONBACKEND_ISPEC TRITONSERVER_Error*
```

Comment:
What does `TRITONBACKEND_ISPEC` do?

Reply:
As I understand it, in Triton, we define TRITONBACKEND_ISPEC as a macro, confusingly, for both `__declspec(dllexport)` and `__declspec(dllimport)`, depending on which module is being compiled ([export](https://learn.microsoft.com/en-us/cpp/build/exporting-from-a-dll-using-declspec-dllexport?view=msvc-170&source=recommendations)/[import](https://learn.microsoft.com/en-us/cpp/build/importing-into-an-application-using-declspec-dllimport?view=msvc-170) docs). These keywords are used to import/export functions, classes, data, etc., from a dll and we need it to import our backend definitions of the functions that core expects. From my understanding, when compiling core, TRITONBACKEND_ISPEC refers to dllimport and imports the functions defined by the backend dll. When compiling the backend, TRITONBACKEND_ISPEC refers to dllexport and exports the implementation of the functions that core expects to the backend dll ([see](https://github.com/triton-inference-server/core/blob/main/include/triton/core/tritonbackend.h#L37) the use of `_COMPILING_TRITONBACKEND`). This macro-defined keyword is necessary for proper functionality on Windows and is used to decorate these common API functions in our backends that support Windows (see [here](https://github.com/triton-inference-server/identity_backend/blob/main/src/identity.cc#L455), for example).

CC @nv-kmcgill53 for resolution of our discussion.

Reply:
Thanks for detailed response.

Reply:
@fpetrini15 sums it up well. The only think I have to add is this other [MS doc page](https://learn.microsoft.com/en-us/cpp/cpp/dllexport-dllimport?view=msvc-170) which discusses this topic some more. One clarification to @fpetrini15's explanation is that it's not about exporting the "implementation" it's about exporting the "symbol name." `dllexport` will export the decorated name (name mangled name), omitting any decoration will cause the function to be `internal` and not visible to the using program, and `cdecl` will export the non-decorated name.  

Comparing this to `.so` files, all symbols are exported by default, so if you can load the `.so` file, you can call the function. I guess I had more to say than I thought....